From ff799022eabbb75e7acc4f668d8433867805540c Mon Sep 17 00:00:00 2001
From: Loren Huang <b02279@freescale.com>
Date: Wed, 20 Nov 2013 14:47:50 +0800
Subject: [PATCH 429/505] ENGR00285547-3 Update mmdc log information

-Update the bus loading description, it alway is overall busloading.
-Add log information when overflow is detected.

Signed-off-by: Loren Huang <b02279@freescale.com>
---
 test/mmdc/mmdc.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/test/mmdc/mmdc.c b/test/mmdc/mmdc.c
index 54357fb..be4f6b7 100644
--- a/test/mmdc/mmdc.c
+++ b/test/mmdc/mmdc.c
@@ -115,7 +115,7 @@ void get_mmdc_profiling_results(pMMDC_t mmdc, MMDC_PROFILE_RES_t *results)
 		results->data_load  	= 0;
 		results->access_utilization	= 0;
 		results->avg_write_burstsize = 0;
-		 results->avg_read_burstsize = 0;
+		results->avg_read_burstsize = 0;
 	}
 }
 void print_mmdc_profiling_results(MMDC_PROFILE_RES_t results, MMDC_RES_TYPE_t print_type, int time)
@@ -146,9 +146,14 @@ void print_mmdc_profiling_results(MMDC_PROFILE_RES_t results, MMDC_RES_TYPE_t pr
 
 		printf("\r\n");
 	}
-	printf("Utilization: %u%%\n",results.utilization);
-	printf("Bus Load: %u%%\n",results.data_load);
-	printf("Bytes Access: %u\n\n",results.access_utilization);
+	if((results.utilization > 100) || (results.data_load > 100))
+	{
+		printf("Warning:Counter overflow!!!Record time too long!!!\n");
+	} else	{
+		printf("Utilization: %u%%\n",results.utilization);
+		printf("Overall Bus Load: %u%%\n",results.data_load);
+		printf("Bytes Access: %u\n\n",results.access_utilization);
+	}
 }
 
 static int get_system_rev(void)
-- 
1.8.0

